sources/api/schnauzer/src/helpers.rs

```diff
@@ -969,77 +983,199 @@ pub fn kube_reserve_cpu(
         })?;
     trace!("Hostname value from template: {}", hostname_value);
 
-    // Create an &str from the serde_json::Value
-    let hostname_str = hostname_value
+    let hosts_value = helper
+        .param(2)
+        .map(|v| v.value())
+        .context(error::InternalSnafu {
+            msg: "Missing params after confirming that they exist.",
+        })?;
+    trace!("Hosts value from template: {}", hosts_value);
+
+    // Extract our variables from their serde_json::Value objects
+    let ip_version = ip_version_value
         .as_str()
         .context(error::InvalidTemplateValueSnafu {
             expected: "string",
-            value: hostname_value.to_owned(),
+            value: ip_version_value.to_owned(),
             template: template_name.to_owned(),
         })?;
-    trace!("Hostname string from template: {}", hostname_str);
+    trace!("IP version string from template: {}", ip_version);
 
-    // Attempt to resolve the hostname
-    let hostname_resolveable = match lookup_host(hostname_str) {
-        Ok(ip_list) => {
-            // If the list of IPs is empty or resolves to localhost, consider the hostname
-            // unresolvable
-            let resolves_to_localhost = ip_list
-                .iter()
-                .any(|ip| ip == &IPV4_LOCALHOST || ip == &IPV6_LOCALHOST);
-            if ip_list.is_empty() || resolves_to_localhost {
-                false
-            } else {
-                true
+    let localhost_comparator = match ip_version {
+        "ipv4" => IPV4_LOCALHOST,
+        "ipv6" => IPV6_LOCALHOST,
+        _ => {
+            return Err(error::TemplateHelperError::InvalidTemplateValue {
+                expected: r#"one of ("ipv4", "ipv6")"#,
+                value: ip_version_value.to_owned(),
+                template: template_name.to_owned(),
             }
-        }
-        Err(e) => {
-            trace!("DNS hostname lookup failed: {},", e);
-            false
+            .into());
         }
     };
 
-    // Only write an entry to the template if the hostname is unresolvable
-    if !hostname_resolveable {
-        let ipv4_entry = format!("{} {}", IPV4_LOCALHOST, hostname_str);
-        let ipv6_entry = format!("{} {}", IPV6_LOCALHOST, hostname_str);
-        let entries = format!("{}\n{}", ipv4_entry, ipv6_entry);
+    let hostname = hostname_value
+        .as_str()
+        .context(error::InvalidTemplateValueSnafu {
+            expected: "string",
+            value: hostname_value.to_owned(),
+            template: template_name.to_owned(),
+        })?;
+    trace!("Hostname string from template: {}", hostname);
+
+    let mut results: Vec<String> = vec![];
 
-        out.write(&entries).context(error::TemplateWriteSnafu {
+    // If our hostname isn't resolveable, add it to the alias list.
+    if hostname.len() > 0 && !hostname_resolveable(hostname) {
+        results.push(hostname.to_owned());
+    }
```

Comment:
I think there's an edge case here where the hostname isn't initially resolvable, but should resolve to a non-localhost IP provided in `settings.network.hosts`, after `/etc/hosts` is populated.

In that case, we'd incorrectly write `/etc/hosts` with the hostname as an alias for localhost.

Reply:
Nice catch! I'll fix that and see if I can't add a unit test to cover that edge case.